Terminal.Gui/Core/Event.cs

```diff
@@ -739,6 +739,12 @@ public struct MouseEvent {
 		/// </summary>
 		public View View;
 
+		/// <summary>
+		/// Indicates if the current mouse event has already been processed and the driver should stop notifying any other event subscriber.
+		/// Its important to set this value to true specially when updating any View's layout from inside the subscriber method.
+		/// </summary>
+		public bool Handled { get; set; }
```

Comment:
In my opinion they should be `pubic getter/setter`. If they are private setters then you must add a constructor to set them. The last prevent changing the properties after they are initialized. But let @tig decide.

Reply:
Ah yeah that's a good point, oops

Reply:
If this change is applied then they are basically the same now.  The only reason for them being seperate that I can see was to add `Handled` and to be a class (instead of a struct) which is now gone (both are classes and both have `Handled`).  The only remaining difference is that `MouseEventArgs` inherts from `EventArgs`.

Not sure how best to phrase that in the docs... other than adding a `<remarks>` that clarifies why the classes are so similar and are not merged (backwards compatibility).

Reply:
But they're not the same.

`MouseEventArgs` *contains* a `MouseEvent` property.

<img width="297" alt="image" src="https://user-images.githubusercontent.com/585482/196713151-902bf1f7-fcc1-4a35-ad5a-f49300d393c9.png">

Is the below accurate?

`MouseEvent` is a low-level construct that conveys the details of mouse events, such as coordinates and button state, from `ConsoleDriver`s up to `Application` and `Views`. The `Application` class includes the `RootMouseEvent` `Action` which takes a `MouseEvent` argument.

`View.MouseEventArgs` is a higher-level construct used for the events defined on `View` and subclasses of `View`  (e.g. `MouseEnter` and `MouseClick`). 

The `Handled` member of `MouseEvent` is only relevant/valid for events that directly use it in their argument definition (e.g. `RootMouseEvent<MouseEvent>`). When using `View.MouseEventArg`-based events, use the `Handled` member of `MouseEventArgs`, not `MouseEvent`.